samza-core/src/main/java/org/apache/samza/startpoint/StartpointSerde.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.startpoint;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import org.apache.samza.SamzaException;
+import org.apache.samza.serializers.Serde;
+import org.codehaus.jackson.map.ObjectMapper;
+
+
+class StartpointSerde implements Serde<Startpoint> {
+  private static final String POSITION_TYPE_FIELD = "positionType";
+  private static final String POSITION_FIELD = "position";
+  private static final String STORED_AT_FIELD = "storedAt";
+
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  @Override
+  public Startpoint fromBytes(byte[] bytes) {
+    try {
+      LinkedHashMap<String, String> deserialized = mapper.readValue(bytes, LinkedHashMap.class);
```

Comment:
1. Why not use the JsonSerde<V2> here for the serialization/deserialization?
2. We've defined custom message format  for startpoint here which is different from our existing message formats: [ContainerLocality](https://github.com/apache/samza/blob/master/samza-core/src/main/java/org/apache/samza/coordinator/stream/messages/SetContainerHostMapping.java), [TaskContainerMapping](https://github.com/apache/samza/blob/master/samza-core/src/main/java/org/apache/samza/coordinator/stream/messages/SetContainerHostMapping.java). I'm okay with it as long as it's json. Just sync up with prateek and see if he has any concerns. Only thing we lose is the consistency across  the different message formats.